internal/db/update.go

```diff
@@ -119,6 +119,60 @@ var updates = map[int]schema.Update{
 	12: updateFromV11,
 	13: updateFromV12,
 	14: updateFromV13,
+	15: updateFromV14,
+}
+
+func updateFromV14(ctx context.Context, tx *sql.Tx) error {
+	_, err := tx.ExecContext(ctx, `CREATE TABLE batches_new (
+    id                            INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+    name                          TEXT NOT NULL,
+    default_target                TEXT NOT NULL,
+    default_target_project        TEXT NOT NULL,
+    status                        TEXT NOT NULL,
+    status_message                TEXT NOT NULL,
+    default_storage_pool          TEXT NOT NULL,
+    include_expression            TEXT NOT NULL,
+    constraints                   TEXT NOT NULL,
+    start_date                    DATETIME NOT NULL,
+    post_migration_retries        INTEGER NOT NULL,
+    rerun_scriptlets              INTEGER NOT NULL,
+    placement_scriptlet           TEXT NOT NULL,
+    restriction_overrides         TEXT NOT NULL,
+    background_sync_interval    INTEGER NOT NULL,
+    final_background_sync_limit INTEGER NOT NULL,
```

Comment:
tiny nit but indent is different on those.

Also, shouldn't the config options just be moved into a JSON TEXT field?
Feels to me like we could have:
 - defaults
   - target
   - target_project
   - storage_pool
 - config
   - post_migration_retries
   - rerun_scriptlets
   - placement_scriptlet
   - restriction_overrides
   - background_sync_interval
   - final_background_sync_limit

That should make it easier to add extra stuff in the future without having a DB schema change to deal with every time.

Reply:
Yeah, following our global/batch/instance config scope discussion I was already planning on doing something like this later, but since it would cut down on a schema update, I've just gone ahead and added these changes here.